### PR TITLE
DOCS Rewrite server requirements and install docs

### PR DESCRIPTION
**Pull request by chillu**
* Move specific guides to wiki entries on forum
* Remove a shitload of noise from the server requirements
* Remove overly specific PHP RNG instructions (that's just built into PHP7
through random_bytes now, which will throw if no suitable RNG is available)
* Remove PHP 5 RNG requirements, since we don't support that PHP release
any mre
* Remove verbose explanation of PHP 5.6 support
* Remove conflicting instructions for PHP memory limits
* Remove version numbers from supporetd databases other than MySQL, it's up
to the community modules to define that
* Remove Oracle support (code is nine years old!)
* Make "community supported" status clearer on databases, people can draw
their own conclusions as open source users on Github
* Remove IIS version number, I think we should just stick to "needs
web.config" and not give the impression that this is actively tested
* Remove mention of OSes for web servers, that's kind of irrelevant in
today's hosting world (containers, PaaS, etc)
* Shorten install instructions in favour of a "quickstart" and point to
lessons instead
* Remove mention of archive download option, we really shouldn't promote
this - composer is the de-facto standard
* Add generic descriptions of the hosting environment considerations
without going too much into specifics
* Remove Apache version number, we don't test on different versions, and
really mostly rely on mod_rewrite working properly. Laravel does the same
(doesn't claim specific Apache version support)

### CODE REVIEW

**Comment by chillu** [comment]
PR for docs redirects: https://github.com/silverstripe/doc.silverstripe.org/pull/191

File: docs/en/00_Getting_Started/00_Server_Requirements.md [hunk 1]
```diff
@@ -4,56 +4,130 @@ SilverStripe CMS needs to be installed on a web server. Content authors and webs
...
-   * Linux/Unix/BSD
-   * Windows
-   * Mac OS X
-
-### PHP Requirements for older SilverStripe releases
+## PHP
+
+ * PHP >=7.1
+ * PHP extensions: `ctype`, `dom`, `fileinfo`, `hash`, `intl`, `mbstring`, `session`, `simplexml`, `tokenizer`, `xml`
+ * PHP configuration: `memory_limit` with at least `48M`
```
Comment (Cheddam) [inline docs/en/00_Getting_Started/00_Server_Requirements.md]: What is this minimum based on? I recall a time where I needed to bump the memory_limit to `256MB` to run a dev/build on a vanilla install, but maybe the memory footprint has improved (particularly in PHP 7.x)?

File: docs/en/00_Getting_Started/00_Server_Requirements.md [hunk 2]
```diff
@@ -4,56 +4,130 @@ SilverStripe CMS needs to be installed on a web server. Content authors and webs
...
+
+### Symlinks
+
+SilverStripe is a modular system, with modules installed and updated
+via the `composer` PHP dependency manager. These are usually stored in `vendor/`,
+outside of the `public/` webroot. Since many modules rely on serving frontend assets
+such as CSS files or images, these are pulled over to the `public/` folder automatically.
+If the filesystem supports it, this is achieved through symlinks.
+Depending on your hosting and deployment mechanisms,
+you might want to enforce copying files instead.
```
Comment (Cheddam) [inline docs/en/00_Getting_Started/00_Server_Requirements.md]:
```suggestion
you may need to configure the plugin to copy files instead.
```

File: docs/en/00_Getting_Started/00_Server_Requirements.md (hunk 1 above)
Comment (chillu) [reply docs/en/00_Getting_Started/00_Server_Requirements.md]: It's what was mentioned as a minimum before in one place, and another place in the same doc said "64M to 512M". In the end, I don't care, it's just taking what's already there. I've seen 4.x requests with ~15M of memory usage, so I think 48M is a safe baseline to communicate.

File: docs/en/00_Getting_Started/00_Server_Requirements.md [hunk 3]
```diff
@@ -66,26 +140,13 @@ SilverStripe's PHP support has changed over time and if you are looking to upgra
...
+## CMS browser requirements
 
 SilverStripe CMS supports the following web browsers:
 * Google Chrome
 * Internet Explorer 11
 * Microsoft Edge 
-* Mozilla Firefox.
+* Mozilla Firefox
  
 We aim to provide satisfactory experiences in Apple Safari. SilverStripe CMS works well across Windows, Linux, and Mac operating systems.
```
[the remark this answers is not in the document]
Comment (chillu) [reply docs/en/00_Getting_Started/00_Server_Requirements.md]: Hmm I think that's worth discussing, but in a separate issue - out of scope for this PR

File: docs/en/00_Getting_Started/00_Server_Requirements.md [hunk 4]
```diff
@@ -4,56 +4,130 @@ SilverStripe CMS needs to be installed on a web server. Content authors and webs
...
+
+SilverStripe is known to work with Microsoft IIS, and generates `web.config` files by default
+(see [Microsoft IIS and SQL Server configuration](https://forum.silverstripe.org/t/microsoft-iis-webserver-and-sql-server-support/2245)).
+
+Additionally, there are community supported guides for installing SilverStripe
+on various environments:
+
+ * [Hosting via Bitnami](https://bitnami.com/stack/silverstripe/virtual-machine): In the cloud or as a locally hosted virtual machine
+ * [Vagrant/Virtualbox with CentOS](https://forum.silverstripe.org/t/installing-via-vagrant-virtualbox-with-centos/2248)
+ * [Mac OSX with Homebrew](https://forum.silverstripe.org/t/installing-on-osx-with-homebrew/2247)
```
[the remark this answers is not in the document]
Comment (chillu) [reply docs/en/00_Getting_Started/00_Server_Requirements.md]: Haha, I thought I was using outdated spelling (after 17 years of using Macs).

File: docs/en/00_Getting_Started/index.md [hunk 5]
```diff
@@ -1,77 +1,53 @@
 title: Getting Started
 introduction: SilverStripe is a web application. This means that you will need to have a webserver and database. We will take you through the setup of the server environment as well the application itself.
 
+## Server Requirements
 
-## Installing SilverStripe
+SilverStripe requires PHP 7.1 or newer.
+It runs on many webservers and databases,
+but feels most at home running on Apache with MySQL/MariaDB.
```
[the remark this answers is not in the document]
Comment (chillu) [reply docs/en/00_Getting_Started/index.md]: Nawww, I liked my stylistic choice for this. But yeah, probably too distracting, yours is clearer

**Comment by ScopeyNZ** [comment]
Is this ready to merge @chillu?

**Pull request merged**